### PR TITLE
Patched EventEmitter for socket.io compatibility with Node >= 7

### PR DESCRIPTION
**Pull request by mserranom**
### Description

As part of Server Pro revival we need to be able to run all the services under the same node version. This is the only service that fails to bootstrap, and this patch fixes it. I haven't tested the behaviour yet, only the bootstrap.

#### Related Issues / PRs

Contributes to https://github.com/overleaf/issues/issues/2043

#### Manual Testing Performed

- [x] service bootstraps locally with Node 10
- [x] service continues working under `dev-environment`

#### Deployment Checklist

- [ ] Check service continues working as usual in staging
- [ ] Check service continues working as usual in production

### CODE REVIEW

File: socket.io.patch.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+// EventEmitter has been removed from process in node >= 7
+// https://github.com/nodejs/node/commit/62b544290a075fe38e233887a06c408ba25a1c71
+if(process.versions.node.split('.')[0] >= 7) {
+  process.EventEmitter = require('events')
+}
+
```
Comment (mserranom) [inline socket.io.patch.js]: I would have liked to include this piece under the `if (io.version === "0.9.16")` block, but `process.EventEmitter` needs to be available before `require("socket.io")`

Comment (briangough) [reply socket.io.patch.js]: I guess `if (!process.EventEmitter) { process.EventEmitter = ... }` is an alternative that doesn't depend on checking the version string.

**Comment by henryoswald** [comment]
Lets hold off on the roll out of this until our other real time work is out and stable (~1 week)

**Comment by mserranom** [comment]
ok, I think I can rely on this branch for Server Pro for the time being.

**Comment by mserranom** [comment]
CC. @briangough @jdleesmiller 
What's the status of the work on real-time stability? Is this PR still a potential merge?

**Comment by henryoswald** [comment]
yep this could go live now.

**Comment by briangough** [comment]
@mserranom I am happy for you to merge this and deploy it.  Let me know if you need any help on the deploy.

**Pull request merged**